presto-main/src/main/java/com/facebook/presto/sql/planner/PlannerUtils.java

```diff
@@ -274,4 +279,27 @@ public static String getPlanString(PlanNode planNode, Session session, TypeProvi
     {
         return PlanPrinter.textLogicalPlan(planNode, types, StatsAndCosts.empty(), metadata.getFunctionAndTypeManager(), session, 0);
     }
+
+    public static boolean isAllLowCardinalityGroupByKeys(AggregationNode aggregationNode, PlanNode source, Metadata metadata, Session session, StatsCalculator statsCalculator, TypeProvider types, long count)
+    {
+        // We already checked that it's just simple scan/filter/project
```

Comment:
can you move this method inside the optimizer itself.  it's making assumptions based on checks the optimizer is doing.

Reply:
this is a useful method for a class of optimizations that I'm doing so I will keep it here Maybe I will rename it but it belongs here and generally useful. FYI, I would have done this optimziation two years ago if this method was here lol. We have to start adding more utilities in the planner. I can rename it maybe if it makes it better

Reply:
Yeah it seems this needs to be called in certain context to make it work. I would move it to respective optimizer, otherwise we should make it more generic.
If the method needs only TableScanNode, it should pass that directly. Getting planNode and then recursively searching for ScanNode seems weird.
Metadata is not used in this method.

Reply:
part of it is the whole back and forth - we shokd have just kept the partitoin key smh. This is gettign annoying

Reply:
Redid this part. see now

Reply:
Looks better now.